### PR TITLE
feat(wd): update to latest upstream release

### PR DESCRIPTION
**Pull request by alpha-tango-kilo**
https://github.com/mfaerevaag/wd/releases/tag/v0.5.2

## Standards checklist:

- [x] The PR title is descriptive.
- [x] The PR doesn't replicate another PR which is already open.
- [x] I have read the contribution guide and followed all the instructions.
- [x] The code follows the code style guide detailed in the wiki.
- [x] The code is mine or it's from somewhere with an MIT-compatible license.
- [x] The code is efficient, to the best of my ability, and does not waste computer resources.
- [x] The code is stable and I have tested it myself, to the best of my abilities.

### CODE REVIEW

File: plugins/wd/wd.plugin.zsh [hunk 1]
```diff
@@ -7,9 +7,4 @@
 #
 # @github.com/mfaerevaag/wd
 
-# Handle $0 according to the standard:
```
Comment (carlosala) [inline plugins/wd/wd.plugin.zsh:10]: We should not remove this to properly handle `$0`. Could you revert this particular change? For the rest, LGTM!

**Review by carlosala — CHANGES_REQUESTED** [review]

File: plugins/wd/wd.plugin.zsh (hunk 1 above)
Comment (alpha-tango-kilo) [reply plugins/wd/wd.plugin.zsh]: Done!

**Comment by carlosala** [comment]
Thanks for your contribution! 😊

**Comment by 0xkag** [comment]
I think there's an issue with this PR.  In my environment, I get:

```
me$ zsh
/path/to/oh-my-zsh/plugins/wd/wd.sh:399: parse error near `]]'
/path/to/oh-my-zsh/plugins/wd/wd.sh:399: parse error near `]]'
/path/to/oh-my-zsh/plugins/wd/wd.sh:399: parse error near `]]'
/path/to/oh-my-zsh/plugins/wd/wd.sh:399: parse error near `]]'
/path/to/oh-my-zsh/plugins/wd/wd.sh:399: parse error near `]]'
```

when I go to launch zsh from bash.

The following patch fixes it:

```diff
diff --git a/plugins/wd/wd.sh b/plugins/wd/wd.sh
index e51cf906..e38f663f 100644
--- a/plugins/wd/wd.sh
+++ b/plugins/wd/wd.sh
@@ -396,7 +396,7 @@ fi
 # disable extendedglob for the complete wd execution time
 setopt | grep -q extendedglob
 wd_extglob_is_set=$?
-[[ $wd_extglob_is_set ]] && setopt noextendedglob
+[[ -z $wd_extglob_is_set ]] && setopt noextendedglob
 
 # load warp points
 typeset -A points
@@ -484,7 +484,7 @@ fi
 # if not, next time warp will pick up variables from this run
 # remember, there's no sub shell
 
-[[ $wd_extglob_is_set ]] && setopt extendedglob
+[[ -z $wd_extglob_is_set ]] && setopt extendedglob
 
 unset wd_extglob_is_set
 unset wd_warp
```

I presume I got the `-z` right, since grep will exit 0 when it finds a match (which you added in the diff on this PR).

@carlosala @alpha-tango-kilo

**Comment by alpha-tango-kilo** [comment]
Hey @0xkag, sorry for breaking your environment! Shoot a PR over onto the wd repo and we'll get it merged and PR'd here for you

**Pull request merged**